sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/RegexpExpressionsSuite.scala

```diff
@@ -323,6 +323,16 @@ class RegexpExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     val nonNullExpr = RegExpReplace(Literal("100-200"), Literal("(\\d+)"), Literal("num"))
     checkEvaluation(nonNullExpr, "num-num", row1)
 
+    // Test empty string replacement
+    val emptyString = RegExpReplace(Literal(""), Literal("^$"), Literal("<empty string>"))
+    checkEvaluation(emptyString, "<empty string>", create_row("", "^$", "<empty string>"))
+    val emptyStringWithPositionOne =
+      RegExpReplace(Literal(""), Literal("^$"), Literal("<empty string>"), 1)
+    checkEvaluation(emptyStringWithPositionOne, "<empty string>", create_row(""))
+    val emptyStringWithPositionGreater =
+      RegExpReplace(Literal(""), Literal("^$"), Literal("<empty string>"), 2)
+    checkEvaluation(emptyStringWithPositionGreater, "", create_row(""))
```

Comment:
These test cases should follows the others.
```
val row7 = create_row("", "^$", "<empty string>")
...
checkEvaluation(expr, "<empty string>", row7)
...
checkEvaluation(exprWithExceedLength, "", row7)
```

Reply:
There is a special case testing for the `nonNullExpr` so I followed that pattern instead of grouping with the other cases, since this is also a special case and wanted to keep it separate.
I am happy to change it though, should I just add `row7` above with the other rows and the `checkEvaluation` bits together with the other `checkEvauation` bits then?

Reply:
OK. we can keep it separate.

Reply:
> I am happy to change it though, should I just add `row7` above with the other rows and the `checkEvaluation` bits together with the other `checkEvauation` bits then?
Looks good. Could you change them and let me see later ?

Reply:
@beliefer I changed the test to be grouped with the others in e2d9948a56495f9affe148661916178e6c5b22a5. Let me know if that's good :)!

Reply:
@LorenzoMartini LGTM although I see it later.